app/models/agents/lifx_pulse_lights_agent.rb

```diff
@@ -0,0 +1,62 @@
+module Agents
+  class LifxPulseLightsAgent < Agent
+    include FormConfigurable
+    
+    cannot_be_scheduled!
+    cannot_create_events!
+
+    description <<-MD
+      Orchestrates your LIFX lights (http://lifx.com/)
+      
+      To get your auth_token:
+      
+      - Log on and go to the Settings page of your [LIFX Account](https://cloud.lifx.com/settings)
+      - Follow the instructions to "Generate New Token"
+      - Copy the new token and use as the `auth_token` -OR- create a new credential called `lifx_auth_token`
+      
+      Read more about the [LIFX HTTP API](http://api.developer.lifx.com/) 
+    MD
+    
+    form_configurable :auth_token
```

Comment:
I initially validated it by making a call to get_lights and quickly found myself rate limited. So then it's just a presence check which didn't add a lot on top of #validate_options. Using a Service as Basecamp does should remove this problem completely.